examples/resize.rs

```diff
@@ -38,7 +41,25 @@ fn main() {
     loop {
         match decoder.read_frame() {
             Ok(frame) => {
-                resizer.run(frame.get_y_plane(), &mut dst);
+                let y_plane = frame.get_y_plane();
+                if bytes_per_sample == 1 {
+                    resizer.resize(y_plane, &mut dst);
+                } else if bytes_per_sample == 2 {
+                    assert_eq!(y_plane.len() % bytes_per_sample, 0);
+                    let y_plane: &[u16] = unsafe {
+                        std::slice::from_raw_parts(
+                            y_plane.as_ptr() as *const u16,
+                            y_plane.len() / bytes_per_sample)
+                    };
+                    let y_plane = match bit_depth {
+                        10 => y_plane.iter().map(|value| (value >> 2) as u8).collect::<Vec<u8>>(),
```

Comment:
resize supports [Gray16](http://docs.piston.rs/resize/resize/Pixel/struct.Gray16.html) pixel type so it should be preferrable to upconvert 10bit to 16bit instead.